vault/core.go

```diff
@@ -236,6 +236,7 @@ type Core struct {
 	// about the seal we are migrating *from*.
 	migrationInfo *migrationInformation
 	sealMigrated  *uint32
+	migrationLock *int32
```

Comment:
May I suggest using Uber's go-atomic wrappers?  That way it's clear how this field is intended to be used.  I might also remove "lock" from the name, calling it "inSealMigration" or something like that, since it's not a mutex which is what we usually think of as locks.

Reply:
Oh, using the wrapper will also protect you against alignment issues on 32-bit platforms.

Reply:
done